orangecontrib/text/widgets/owscoredocuments.py

```diff
@@ -0,0 +1,572 @@
+            for c in re.split("([0-9]+)", key)
+        ]
+
+    def lessThan(self, left_ind, right_ind):
+        """
+        Sort strings of the first column naturally: Document 2 < Document 12
+        """
+        if left_ind.column() == 0 and right_ind.column() == 0:
+            left = self.sourceModel().data(left_ind, role=Qt.DisplayRole)
+            right = self.sourceModel().data(right_ind, role=Qt.DisplayRole)
+            return self._alphanum_key(left) < self._alphanum_key(right)
+        return super().lessThan(left_ind, right_ind)
+
+
+class ScoreDocumentsTableModel(PyTableModel):
+    def data(self, index, role=Qt.DisplayRole):
+        if role in (gui.BarRatioRole, Qt.DisplayRole):
+            dat = super().data(index, Qt.EditRole)
+            return dat
+        if role == Qt.BackgroundColorRole and index.column() == 0:
+            return TableModel.ColorForRole[TableModel.Meta]
+        return super().data(index, role)
+
+
+class OWScoreDocuments(OWWidget, ConcurrentWidgetMixin):
+    name = "Score Documents"
+    description = ""
+    icon = "icons/ScoreDocuments.svg"
+    priority = 500
+
+    auto_commit: bool = Setting(True)
+    aggregation: int = Setting(0)
+
+    word_frequency: bool = Setting(True)
+    word_appearance: bool = Setting(False)
+    embedding_similarity: bool = Setting(False)
+    embedding_language = Setting(0)
+
+    class Inputs:
+        corpus = Input("Corpus", Corpus)
+        words = Input("Words", Table)
+
+    class Outputs:
+        corpus = Output("Corpus", Corpus)
+
+    class Warning(OWWidget.Warning):
+        missing_words = Msg("Provide words on the input")
+        missing_corpus = Msg("Provide corpus on the input")
+        corpus_not_normalized = Msg("Use Preprocess Text to normalize corpus.")
+
+    class Error(OWWidget.Error):
+        unknown_err = Msg("{}")
+
+    def __init__(self):
+        OWWidget.__init__(self)
+        ConcurrentWidgetMixin.__init__(self)
+        self._setup_control_area()
+        self._setup_main_area()
+        self.corpus = None
+        self.words = None
+        # saves scores avoid multiple computation of the same score
+        self.scores = {}
+
+    def _setup_control_area(self) -> None:
+        box = gui.widgetBox(self.controlArea, "Word scoring")
+        for value, (n, _, tt) in SCORING_METHODS.items():
+            b = gui.hBox(box, margin=0)
+            gui.checkBox(
+                b,
+                self,
+                value,
+                label=n,
+                callback=self.__setting_changed,
+                tooltip=tt,
+            )
+            if value in ADDITIONAL_OPTIONS:
+                value, options = ADDITIONAL_OPTIONS[value]
+                gui.comboBox(
+                    b,
+                    self,
+                    value,
+                    items=options,
+                    callback=self.__setting_changed,
+                )
+
+        box = gui.widgetBox(self.controlArea, "Aggregate scores")
```

Comment:
"Aggregate scores" -> "Aggregation" to make it consistent with the `Extract Keywords` widget